### PR TITLE
Change proxyupdater to use model config service

### PR DESCRIPTION
**Pull request by barrettj12**
Modify the proxyupdater agent facade to use a model config service instead of getting model config from state.

Includes corresponding test changes. I've modified the tests to use gomock-generated mocks for both `ControllerConfigService` and `ModelConfigService`, as per agreed convention.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- ~[ ] Comments saying why design decisions were made~
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Run unit tests.

Manual QA steps:

```console
$ juju bootstrap lxd c
$ juju switch controller

# Check existing proxy settings on disk
$ juju exec --machine 0 -- cat /etc/juju-proxy.conf

# Change proxy settings in model config
$ juju model-config http-proxy='foo'

# Check proxy settings have been updated
$ juju exec --machine 0 -- cat /etc/juju-proxy.conf
export http_proxy=foo
export HTTP_PROXY=foo
export no_proxy=...
export NO_PROXY=...
```

## Links

**Jira card:** JUJU-6057

### CODE REVIEW

File: apiserver/facades/agent/proxyupdater/proxyupdater_test.go [hunk 1]
```diff
@@ -53,13 +57,18 @@ func (s *ProxyUpdaterSuite) SetUpTest(c *gc.C) {
 	s.state.SetUp(c)
 	s.AddCleanup(func(_ *gc.C) { s.state.Kill() })
 
-	api, err := proxyupdater.NewAPIV2(s.state, s.state, &stubControllerConfigService{}, s.resources, s.authorizer)
+	s.modelConfigService = &stubModelConfigService{}
```
Comment (hpidcock) [inline apiserver/facades/agent/proxyupdater/proxyupdater_test.go:60]: Looking at the facade code, I think moving to mocks would be trivial. Can we please change to that?

**Review by hpidcock — CHANGES_REQUESTED** [review]

**Comment by hpidcock** [comment]
Please add QA steps that test changing proxy settings and seeing them change on an agent.

File: apiserver/facades/agent/proxyupdater/proxyupdater.go [hunk 2]
```diff
@@ -32,46 +33,40 @@ type ControllerConfigService interface {
 	ControllerConfig(context.Context) (controller.Config, error)
 }
 
+// ModelConfigService provides access to the model's configuration.
+type ModelConfigService interface {
```
Comment (tlm) [inline apiserver/facades/agent/proxyupdater/proxyupdater.go:37]: Can we please move this into a `services.go` file and also add comments for the functions. Comments can be copied from the service.

**Review by tlm — COMMENTED** [review]
Thank you for the work. When you are ready please updating testing steps and drag this PR out of draft. Also Jira card into current sprint.

**Review by hpidcock — APPROVED** [review]
Thanks for the QA steps. Manually testing passed.

**Review by tlm — APPROVED** [review]
lgtm

**Pull request merged**